ts/parts-map/ColorSeriesMixin.ts

```diff
@@ -106,7 +107,7 @@ H.colorSeriesMixin = {
 
             color = point.options.color ||
                 (
-                    point.isNull ?
+                    point.isNull || point.value === null ?
```

Comment:
This addition shouldn't be necessary. When `point.value` is `null`, `point.isNull` should always be true. See if this can be handled in the `isValid` function for the given Point class.

Reply:
Moving above condition into `isValid` function does not give any effect, because if `point.value` will be null, then the point graphic (marker) will not be rendered at all, as it is in case of scatter series on which a heatmap is based on since refactoring.

Before implementing the ability to customise heatmap markers, this series used `drawPoints` method from column series, where the flag `point.isNull` is not checked during point rendering process, but only whether the `point.y` is not equal to `null`. That’s the reason of why heatmap points with` point.value = null` had graphics before, and most probably we should keep it to work in the same way in the new implementation.

We can either override `Series.drawPoints` function and move a whole logic into the new function, but with only one change in if statement (what does not look as a good way out), or keep the current implementation which does work correctly, but can be a bit difficult to understand at first glance.
We can also consider other solutions, so if you would have any ideas or suggestions of how to resolve it, will be great to discuss about it.